plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -136,23 +141,51 @@ compileLiteral = \case
 -- do different things to the inner expression. This one assumes it's a literal, the other one keeps compiling
 -- through it.
 
--- | Get the bytestring content of a string expression, if possible. Follows (Haskell) variable references!
-stringExprContent :: GHC.CoreExpr -> Maybe BS.ByteString
-stringExprContent = \case
-  GHC.Lit (GHC.LitString bs) -> Just bs
-  -- unpackCString# / unpackCStringUtf8# are just wrappers around a literal
-  GHC.Var n `GHC.App` expr
-    | let name = GHC.getName n
-    , name == GHC.unpackCStringName || name == GHC.unpackCStringUtf8Name ->
-      stringExprContent expr
+data StringExprContentAs = AsBytes | AsText
```

Comment:
The `stringExprContent` function is used to extract bytes for the `BuiltinByteString` as well as for the `BuiltinString` literals; 
I have added this datatype as its parameter to distinguish between these scenarios as we want to utf-8 decode `BuiltinByteString` literals only.